### ISSUE TITLE
Can't mount tmpfs

### ISSUE THREAD

**Issue by Jehan** [flatpak/flatpak#857]
After building GIMP flatpak, installing and trying to run it:

> $ flatpak run org.gimp.GIMP 
> Can't mount tmpfs on /newroot/home/jehan/.local/share/flatpak: No such file or directory
> $ flatpak --version
> Flatpak 0.9.5

This is the flatpak from Fedora 25.
I have not changed anything in our manifest which used to work not long ago: https://git.gnome.org/browse/gimp/tree/build/flatpak/org.gimp.GIMP.json
So I imagine the problem is that something changed in flatpak?

**Comment by Jehan** [comment]
Actually I have this issue with any of the flatpak I had installed (and which used to work). I just can't run them anymore. That's not specific to our GIMP flatpak. What's wrong with my environment? I tried to delete .cache/flatpak/ and .local/share/flatpak/ thinking I would reset the full db of installed flatpak-s but it doesn't change a thing.

**Comment by Jehan** [comment]
So I created a test user with an empty home, and it can install and run a flatpak no problem. So the problem is definitely something rotten in my user environment.
Could you tell me what flatpak installs other than .cache/flatpak/ and .local/share/flatpak/? I don't mind just deleting everything I installed locally with flatpak and reinstalling to get a working flatpak.

**Comment by alexlarsson** [comment]
Weird. Does `/home/jehan/.local/share/flatpak` exist? It seems to fail when trying to mount a tmpfs at this position, which it does in order to hide any per-user installed applications from other apps.

**Comment by Jehan** [comment]
>  Does /home/jehan/.local/share/flatpak exist?

It does exist, belongs to my user with write access.

**Comment by alexlarsson** [comment]
That is really weird. Are any of the subpaths of that a symlink?

**Comment by Jehan** [comment]
No.

**Comment by Jehan** [comment]
Could you please tell me what folders to delete to just get a clean slate?
I don't even care about fixing this bug anymore, or finding the issue. I just want to be able to run flatpak builds. I did not remember why I had not been updating our GIMP flatpak for the last few months. Now I do: because it doesn't work on my environment. ;-(

Right now, I have to log in as another user (on the same laptop/OS, so that's nearly definitely some user environment issue) to be able to run a flatpak build.
As said previously, I have already deleted .cache/flatpak/ and .local/share/flatpak/. What else is there to delete?

**Comment by alexlarsson** [comment]
Is /home/jehan by any chance a symlink?

**Comment by Jehan** [comment]
No.
/home/jehan/.cache used to be a symlink, but I got rid of it to make sure, deleted .cache/flatpak/ (as well as .local/share/flatpak), reinstalled the flatpak. Yet it still returns this error.
I don't see any of symlink which I could have created there.

**Comment by alexlarsson** [comment]
I mean /home/jehan itself.

**Comment by alexlarsson** [comment]
This sounds *so* like https://github.com/flatpak/flatpak/issues/1032, but you're sure  ~/.local/share/flatpak is not a symlink?

**Comment by Jehan** [comment]
I checked again. There is absolutely no symlink in any component of this path. Neither ~/.local/share/flatpak, nor ~/.local/share/ nor ~/.local/ nor ~ (/home/jehan), nor /home.

**Comment by alexlarsson** [comment]
Do you have some uncommon env-var set? `XDG_DATA_HOME` for instance?

**Comment by Jehan** [comment]
Well I have a lot of environment variable in my bashrc so yes. I don't have `XDG_DATA_HOME` though, but I have some other `XDG_*`. For instance I have `XDG_CACHE_HOME` and `XDG_CONFIG_HOME` but they are set to their default values. The only reason I explicitly set them in the bashrc was to have access to them in other programs which didn't know of XDG standard but could read environment variables (in particular this was so that my vim config files would be saved under `XDG_CONFIG_HOME`).
Also this has been there likely for a very long time.

**Comment by alexlarsson** [comment]
Can you try building flatpak master? I just changes some behaviour wrt this.
If you have a packaged flatpak you can just clone git and build it, then run `./flatpak run ..` in-tree. No need to install.

**Comment by Jehan** [comment]
P.S.: I'll try to make some tests to see if my bashrc is the culprit. But this will have to wait for a few days because it needs some prep to not completely fuck up my environment.
I have a lot of software tweaked for instance to save inside $XDG_CONFIG_HOME (even when they don't do it by default). And if I just remove (even temporarily) my bashrc, that will create instantly tons of hidden folders and files in my HOME. But I'll try after some preparation.

**Comment by Jehan** [comment]
> Can you try building flatpak master? I just changes some behaviour wrt this.

Ok will do now.

**Comment by alexlarsson** [comment]
Also, no need to change bashrc. Just start a shell and manually unset the variables.

**Comment by Jehan** [comment]
Sorry took longer than expected, I also had to build ostree since Fedora 25 doesn't have a recent enough version for flatpak.

> No need to install.

For info, I still had to install, otherwise flatpak failed to load because it could not run `libexec/flatpak-bwrap`.

> I just changes some behaviour wrt this.

In any cases, yes it works with master flatpak!
Checking a bit the commits, it has been fixed by commit ac5f17203f8b51a09aaf2edbdea5b3a539ee5063.
So surprised, I rechecked and want to say I am very very sorry to have made debugging more complicated:

> $ ls -dltr .local/share/flatpak/
> drwx------. 8 jehan jehan 4096 Sep 25 16:10 .local/share/flatpak/
> $ ls -dltr .local/share/flatpak
> lrwxrwxrwx. 1 jehan jehan 41 Jun 12 03:45 .local/share/flatpak -> /opt/user-cache/jehan/local_share_flatpak

I did have a symlink! But when checking with the finale slash (auto-completion!), it was not showing, even though using -d option!
Really again, sorry.
